File: JsonSchemaAndStructs/KiteConnect-Kotlin-Structs/gtt_modify_order.kt
// To parse the JSON, install Klaxon and do:
//
//   val gttModifyOrder = GttModifyOrder.fromJson(jsonString)

package quicktype

import com.beust.klaxon.*

private val klaxon = Klaxon()

data class GttModifyOrder (
    @Json(name = "\$ref")
    val ref: String,

    @Json(name = "\$schema")
    val schema: String,

    val definitions: Definitions
) {
    public fun toJson() = klaxon.toJsonString(this)

    companion object {
        public fun fromJson(json: String) = klaxon.parse<GttModifyOrder>(json)
    }
}

data class Definitions (
    @Json(name = "Data")
    val data: Data,

    @Json(name = "GttModifyOrder")
    val gttModifyOrder: GttModifyOrderClass
)

data class Data (
    val additionalProperties: Boolean,
    val properties: DataProperties,
    val required: List<String>,
    val title: String,
    val type: String
)

data class DataProperties (
    @Json(name = "trigger_id")
    val triggerID: TriggerID
)

data class TriggerID (
    val type: String
)

data class GttModifyOrderClass (
    val additionalProperties: Boolean,
    val properties: GttModifyOrderProperties,
    val required: List<String>,
    val title: String,
    val type: String
)

data class GttModifyOrderProperties (
    val data: DataClass,
    val status: TriggerID
)

data class DataClass (
    @Json(name = "\$ref")
    val ref: String
)

File: JsonSchemaAndStructs/KiteConnect-Rust-Structs/gtt_modify_order.rs
// Example code that deserializes and serializes the model.
// extern crate serde;
// #[macro_use]
// extern crate serde_derive;
// extern crate serde_json;
//
// use generated_module::[object Object];
//
// fn main() {
//     let json = r#"{"answer": 42}"#;
//     let model: [object Object] = serde_json::from_str(&json).unwrap();
// }

extern crate serde_derive;

#[derive(Serialize, Deserialize)]
pub struct GttModifyOrder {
    #[serde(rename = "$ref")]
    gtt_modify_order_ref: String,

    #[serde(rename = "$schema")]
    schema: String,

    #[serde(rename = "definitions")]
    definitions: Definitions,
}

#[derive(Serialize, Deserialize)]
pub struct Definitions {
    #[serde(rename = "Data")]
    data: Data,

    #[serde(rename = "GttModifyOrder")]
    gtt_modify_order: GttModifyOrderClass,
}

#[derive(Serialize, Deserialize)]
pub struct Data {
    #[serde(rename = "additionalProperties")]
    additional_properties: bool,

    #[serde(rename = "properties")]
    properties: DataProperties,

    #[serde(rename = "required")]
    required: Vec<String>,

    #[serde(rename = "title")]
    title: String,

    #[serde(rename = "type")]
    data_type: String,
}

#[derive(Serialize, Deserialize)]
pub struct DataProperties {
    #[serde(rename = "trigger_id")]
    trigger_id: TriggerId,
}

#[derive(Serialize, Deserialize)]
pub struct TriggerId {
    #[serde(rename = "type")]
    trigger_id_type: String,
}

#[derive(Serialize, Deserialize)]
pub struct GttModifyOrderClass {
    #[serde(rename = "additionalProperties")]
    additional_properties: bool,

    #[serde(rename = "properties")]
    properties: GttModifyOrderProperties,

    #[serde(rename = "required")]
    required: Vec<String>,

    #[serde(rename = "title")]
    title: String,

    #[serde(rename = "type")]
    gtt_modify_order_class_type: String,
}

#[derive(Serialize, Deserialize)]
pub struct GttModifyOrderProperties {
    #[serde(rename = "data")]
    data: DataClass,

    #[serde(rename = "status")]
    status: TriggerId,
}

#[derive(Serialize, Deserialize)]
pub struct DataClass {
    #[serde(rename = "$ref")]
    data_class_ref: String,
}
